core/commands/blame.py

```diff
@@ -238,147 +320,175 @@ def couple_partitions_and_commits(partitions, commit_infos, left_pad):
+
+    def perpare_and_apply_patched_theme(self):
+        """
+        self.view.add_regions has a bug (may be feature) of not apply the scope color
+        if the region has the same color of background as the applied scope.
+
+        A patched theme is created with an additional scope BLAME_SCOPE.
+        This additional scope will color all the texts in the blame view will
+        a color which offsets from the original backgound color by a little bit.
+        """
+        if self._original_color_scheme:
+            # it means the color scheme has been patched
+            return
 
+        self._original_color_scheme = self.view.settings().get("color_scheme")
 
-class GsBlameNavigateChunkCommand(GsNavigate):
-
-    """
-    Move cursor to the next (or previous) different commit
-    """
-
-    offset = 0
-
-    def get_available_regions(self):
-        return [
-            branch_region
-            for region in self.view.find_by_selector(
-                "constant.numeric.commit-hash.git-savvy"
-            )
-            for branch_region in self.view.lines(region)]
-
+        if "GitSavvy.blame" in self._original_color_scheme:
+            return
 
-class GsBlameActionCommand(PanelActionMixin, TextCommand, GitCommand):
+        if self._theme:
+            self.view.settings().set("color_scheme", self._theme)
+        else:
+            theme = ThemeGenerator(self._original_color_scheme)
+            background_color = theme.get_global_settings("background")
+            theme.add_scoped_style(
+                "Git Savvy Blame",
+                BLAME_SCOPE,
+                background=util.color.next_color(background_color))
+            self._theme = os.path.join(
+                "Packages", theme.get_new_theme_path("blame")).replace("\\", "/")
+            theme.apply_new_theme("blame", self.view)
+
+    def highlight(self, commit_hash):
+        """
+        Create an output view then copy the scopes and apply to the blame view.
+        """
+
+        content = self.get_file_content_at_commit(self.file_path, commit_hash)
+        output_view = self.view.window().create_output_panel("GitSavvy Blame")
+        output_view.run_command("gs_new_content_and_regions", {
+                    "content": content,
+                    "regions": {},
+                    "nuke_cursors": False
+                })
+        output_view.settings().set(
+            "syntax", self.view.settings().get('git_savvy.original_syntax', None))
+
+        blame_view_pt = 0  # a start pointer to search for line number
+        flags = sublime.DRAW_NO_OUTLINE | sublime.PERSISTENT
+
+        count = 0
+        for lineno, line in enumerate(output_view.lines(sublime.Region(0, output_view.size()))):
+            if not line.empty():
+                pattern = r".{{30}} \| {lineno: >4}\s".format(lineno=lineno+1)
+                corresponding_region = self.view.find(pattern, blame_view_pt)
```

Comment:
No problem. Btw, come back to your questions, we although know which column the code starts, but the line numbers of the original code gives on information of the line numbers of the corresponding lines for the blame view. For example, line 3 of the original file is line 5 of the blame view.

<img width="726" alt="screen shot 2017-08-10 at 2 53 09 am" src="https://user-images.githubusercontent.com/1690993/29157969-17712ef2-7d77-11e7-8601-e2ccdbbbe227.png">